eva-server/src/main/java/uk/ac/ebi/eva/server/ws/RegionWSServerV2.java

```diff
@@ -137,7 +142,49 @@ public ResponseEntity getVariantsByRegion(@PathVariable("regionId") String regio
 
             resourcesList.add(new Resource<>(variant, Arrays.asList(sourcesLink, annotationsLink)));
         });
-        return new ResponseEntity(new Resources<>(resourcesList), HttpStatus.OK);
+
+        PagedResources pagedResources = new PagedResources<>(resourcesList, pageMetadata);
+
+        if (pageable.getPageNumber() != 0) {
+            pagedResources.add(new Link(linkTo(methodOn(RegionWSServerV2.class).getVariantsByRegion(regionId, species,
+                    assembly, studies, consequenceType, maf, polyphenScore, siftScore, annotationVepVersion,
+                    annotationVepCacheVersion, new PageRequest(pageable.getPageNumber() - 1, pageable.getPageSize()),
+                    response, request))
+                    .toUriComponentsBuilder()
+                    .queryParam("size", pageable.getPageSize())
+                    .queryParam("page", pageable.getPageNumber() - 1)
+                    .toUriString(), "prev"));
+
+            pagedResources.add(new Link(linkTo(methodOn(RegionWSServerV2.class).getVariantsByRegion(regionId, species,
+                    assembly, studies, consequenceType, maf, polyphenScore, siftScore, annotationVepVersion,
+                    annotationVepCacheVersion, new PageRequest(0, pageable.getPageSize()),
+                    response, request))
+                    .toUriComponentsBuilder()
+                    .queryParam("size", pageable.getPageSize())
+                    .queryParam("page", 0)
+                    .toUriString(), "first"));
+        }
+        if (pageable.getPageNumber() + 1 != pageMetadata.getTotalPages()) {
+            pagedResources.add(new Link(linkTo(methodOn(RegionWSServerV2.class).getVariantsByRegion(regionId, species,
+                    assembly, studies, consequenceType, maf, polyphenScore, siftScore, annotationVepVersion,
+                    annotationVepCacheVersion, new PageRequest(pageable.getPageNumber() + 1, pageable.getPageSize()),
+                    response, request))
+                    .toUriComponentsBuilder()
+                    .queryParam("size", pageable.getPageSize())
+                    .queryParam("page", pageable.getPageNumber() + 1)
+                    .toUriString(), "next"));
+
+            pagedResources.add(new Link(linkTo(methodOn(RegionWSServerV2.class).getVariantsByRegion(regionId, species,
+                    assembly, studies, consequenceType, maf, polyphenScore, siftScore, annotationVepVersion,
+                    annotationVepCacheVersion, new PageRequest(
+                            (int)pageMetadata.getTotalPages()-1, pageable.getPageSize()),
+                    response, request))
+                    .toUriComponentsBuilder()
+                    .queryParam("size", pageable.getPageSize())
```

Comment:
extract at the very minimum those strings that appear several times. The strings that appear in this class once but are used from other classes should arguably be extracted too.

Reply:
I have changed `size` and `page` to constant Strings :+1: